blueocean-bitbucket-pipeline/src/main/java/io/jenkins/blueocean/blueocean_bitbucket_pipeline/BitbucketCredentialUtils.java

```diff
@@ -0,0 +1,22 @@
+package io.jenkins.blueocean.blueocean_bitbucket_pipeline;
+
+import com.google.common.base.Preconditions;
+import io.jenkins.blueocean.blueocean_bitbucket_pipeline.cloud.BitbucketCloudScm;
+import io.jenkins.blueocean.blueocean_bitbucket_pipeline.server.BitbucketServerScm;
+import org.apache.commons.codec.digest.DigestUtils;
+
+/**
+ * @author cliffmeyers
+ */
+public class BitbucketCredentialUtils {
+
+    public static String computeCredentialId(String scmId, String apiUrl) {
```

Comment:
It should also take credentialId as parameter so that it knows fallback, otherwise everywhere we are doing null check for credentialId then calling it.